model/shared/src/main/scala/explore/data/KeyedIndexedList.scala

```diff
@@ -99,6 +99,10 @@ case class KeyedIndexedList[K, A] private (private val list: TreeSeqMap[K, (A, N
   def updatedValueWith(key: K, f: A => A): KeyedIndexedList[K, A] =
     updatedWith(key, (v, i) => (f(v), i))
 
+  // WARNING - Do not use to update the part of the value used as a key
+  def unsafeMapValues(f: A => A): KeyedIndexedList[K, A] =
```

Comment:
You could make this safe by passing a lens for the key so that it gets reinstated in case it's modified. In fact, this would be automatic if you use `KIListMod(keyLens).mod`

Reply:
While digging through the KeyedIndexexList stuff to figure out how to make this work, I realized where I'm using this isn't participating in `undo`. 😢 I'll take care of that, and then figure this out.

Reply:
Actually, I'm not sure this should participate in `undo`. We can't get rid of a ConfigurationRequest via the API. I guess once we get a mutation we could Withdraw the request on `undo` and set it back to `Requested` on `redo`? It's not quite the same as fully undoing it.

Reply:
Sounds like we want to make those actions explicit. Like submit/retract a proposal.

Reply:
Yeah. I think you're right. Undo doesn't seem appropriate here. Thanks for the input.

Reply:
I'll deal with the unsafeMapValues in a new PR. I want to get this merged so Rob can continue cleaning up the odb/core.